apps/omg_watcher/lib/exit_processor/core.ex

```diff
@@ -502,6 +513,40 @@ defmodule OMG.Watcher.ExitProcessor.Core do
     {chain_validity, events}
   end
 
+  @spec get_exit_info(Utxo.Position.t(), t()) :: {:ok, ExitInfo.t()} | {:ok, :not_found}
+  def get_exit_info(Utxo.position(_, _, _) = utxo_exit, %__MODULE__{exits: exits}) do
+    case Map.get(exits, utxo_exit) do
+      nil -> {:ok, :not_found}
+      exit_info -> {:ok, exit_info}
+    end
+  end
+
+  @spec get_ife_based_on_utxo(Utxo.Position.t(), t()) :: {:ok, Transaction.Signed.t()} | {:ok, :not_found}
+  def get_ife_based_on_utxo(Utxo.position(_, _, _) = utxo_exit, %__MODULE__{} = state) do
+    get_known_txs(state)
+    |> Enum.find(fn %KnownTx{signed_tx: %Transaction.Signed{raw_tx: %Transaction{} = tx}} ->
+      Enum.member?(Transaction.get_inputs(tx), utxo_exit)
+    end)
+    |> case do
+      nil -> {:ok, :not_found}
+      signed_tx -> {:ok, signed_tx}
+    end
+  end
+
+  @spec get_invalid_exits_based_on_ifes(t()) :: list(%{Utxo.Position.t() => ExitInfo.t()})
+  defp get_invalid_exits_based_on_ifes(%__MODULE__{exits: exits} = state) do
+    exiting_utxo_positions =
+      get_known_txs(state)
+      |> Enum.flat_map(fn %KnownTx{signed_tx: %Transaction.Signed{raw_tx: %Transaction{} = tx}} ->
+        Transaction.get_inputs(tx)
+      end)
+
+    exits
+    |> Enum.filter(fn {utxo_pos, _exit_info} ->
+      Enum.find(exiting_utxo_positions, fn exiting_utxo_pos -> utxo_pos == exiting_utxo_pos end)
```

Comment:
ther is Kernel.match?

Reply:
result using `match?`;
`{true, {:utxo_position, 9000, 0, 1}, {:utxo_position, 1, 0, 0}}`
When using `match?` {:utxo_position, 9000, 0, 1} equals {:utxo_position, 1, 0, 0} I think because there is macro or somethig .
In the console(IEX) it will be true.